### PR TITLE
Fixed java client's implementation of NetworkConnection.

### PR DESCRIPTION
**Pull request by mcharsley**
- [ X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

Changed java client's RemoteNetworkConnection to send just the integer
bitmask for the network connection, not the full object.
This will bring the java client in line with
 - the python client (see https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/mobile.py#L60 )
 - what the spec expects (see https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#device-modes )
 - and what chromedriver expects. prior to this change I saw

    [18.564][INFO]: COMMAND SetNetworkConnection {
       "parameters": {
          "type": {
             "airplaneMode": true,
             "class": "org.openqa.selenium.mobile.NetworkConnection$ConnectionType",
             "dataEnabled": false,
             "hCode": 1,
             "wifiEnabled": false
          }
       }
    }
being sent, which resulted in a RESPONSE SetNetworkConnection unknown
error: invalid connection_type error.

### CODE REVIEW

File: java/client/src/org/openqa/selenium/remote/mobile/RemoteNetworkConnection.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public ConnectionType getNetworkConnection() {
   @Override
   public ConnectionType setNetworkConnection(
       ConnectionType type) {
-    Map<String, ConnectionType> mode = ImmutableMap.of("type", type);
+    Map<String, Integer> mode = ImmutableMap.of("type", type.getBitMask());
```
Comment (lukeis) [inline java/client/src/org/openqa/selenium/remote/mobile/RemoteNetworkConnection.java]: can you change this instead to just `type.toString()` and then you wouldn't have to expose the getBitMask in the enum. (Alternatively you could have used `type.hashCode()` but that doesn't feel as nice)

Comment (lukeis) [inline java/client/src/org/openqa/selenium/remote/mobile/RemoteNetworkConnection.java]: looking at it again, we can just add the `toJson()` method to ConnectionType and then the JsonToBeanConverter will do it's thing and serialize it.

    public String toJson(){ return toString(); }

Comment (mcharsley) [inline java/client/src/org/openqa/selenium/remote/mobile/RemoteNetworkConnection.java]: Reflection!? Oh well, when in Rome... Have removed getBitMask, and added toJson method. Thanks.

**Comment by mcharsley** [comment]
Lukeis: I've pushed some new changes corresponding to your comments
Being completely new to github, I'm not sure if you automatically get informed or not. Hopefully if you haven't then this will let you know...

Please take another look? thanks

**Pull request merged**